File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
  def romanToInt(self, s: str) -> int:
    lookup = {
      'I': 1,
      'V': 5,
      'X': 10,
      'L': 50,
      'C': 100,
      'D': 500,
      'M': 1000,
    }

    ans = i = 0
    while i < len(s):
      if i+1 == len(s) or lookup[s[i]] >= lookup[s[i+1]]:
        ans += lookup[s[i]]
        i += 1
      else:
        ans += lookup[s[i+1]] - lookup[s[i]]
        i += 2

    return ans

File: 0013-roman-to-integer/0013-roman-to-integer.go
func romanToInt(s string) int {
  lookup := map[string]int{
    "I": 1,
    "V": 5,
    "X": 10,
    "L": 50,
    "C": 100,
    "D": 500,
    "M": 1000,
  }
  
  ans, i := 0, 0
  for i < len(s) {
    if i < len(s)-1 && lookup[s[i:i+1]] < lookup[s[i+1:i+2]] {
      ans += lookup[s[i+1:i+2]]-lookup[s[i:i+1]]
      i += 2
    } else {
      ans += lookup[s[i:i+1]]
      i++
    }
  }
  return ans
}